internal/flix/flix.go

```diff
@@ -0,0 +1,181 @@
+package flix
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/onflow/flow-cli/internal/scripts"
+
+	"github.com/onflow/flow-cli/internal/transactions"
+
+	"github.com/onflow/flixkit-go"
+	"github.com/onflow/flow-cli/flowkit"
+	"github.com/onflow/flow-cli/flowkit/output"
+	"github.com/onflow/flow-cli/internal/command"
+	"github.com/spf13/cobra"
+)
+
+type Flags struct {
+	ArgsJSON    string   `default:"" flag:"args-json" info:"arguments in JSON-Cadence format"`
+	BlockID     string   `default:"" flag:"block-id" info:"block ID to execute the script at"`
+	BlockHeight uint64   `default:"" flag:"block-height" info:"block height to execute the script at"`
+	Signer      string   `default:"" flag:"signer" info:"Account name from configuration used to sign the transaction as proposer, payer and suthorizer"`
+	Proposer    string   `default:"" flag:"proposer" info:"Account name from configuration used as proposer"`
+	Payer       string   `default:"" flag:"payer" info:"Account name from configuration used as payer"`
+	Authorizers []string `default:"" flag:"authorizer" info:"Name of a single or multiple comma-separated accounts used as authorizers from configuration"`
+	Include     []string `default:"" flag:"include" info:"Fields to include in the output"`
+	Exclude     []string `default:"" flag:"exclude" info:"Fields to exclude from the output (events)"`
+	GasLimit    uint64   `default:"1000" flag:"gas-limit" info:"transaction gas limit"`
+}
+
+var flags = Flags{}
+
+var Cmd = &cobra.Command{
+	Use:   "flix",
+	Short: "Commands for the Flix functionality",
+}
+
+var idCommand = &command.Command{
```

Comment:
by using flags you could merge all three functions into one. And then just handle based on the flag present.

Reply:
Addressed